pages/verification/index.tsx

```diff
@@ -32,21 +34,74 @@ export default function Verification(props: PageProps) {
     const initChatAnon = useInitChatAnonymousMutation();
     const [isLoading, setIsLoading] = useState(props.exchangeToken ? true : false)
     const getLoginPage = () => {
+        sendAnalytics(BetterSocialEventTracking.VERIFICATION_SCREEN_HUMAN_ID_BUTTON_CLICKED)
         fetch('/api/getWebLogin').then(async (res) => {
             toast('Sending you to humanID for verification', {
                 autoClose: false,
                 type: 'info',
             })
             const data = await res.json();
+            console.log('data', data);
             if (data.data?.webLoginUrl) {
                 router.push(data.data?.webLoginUrl)
             }
         })
     }
 
+    const sendAnonymousMessage = (anonymousToken: string) => {
+        const member = localStorage.getItem(MessageEnum.targetUser);
+        const message = localStorage.getItem(MessageEnum.tempMessage);
+        apiAnonymous.defaults.headers.common['Authorization'] = `Bearer ${anonymousToken}`;
+        generateAnonUser.mutate({ userId: member }, {
+            onSuccess: (data) => {
+                initChatAnon.mutate({
+                    anon_user_info_color_code: data.anon_user_info_color_code,
+                    anon_user_info_color_name: data.anon_user_info_color_name,
+                    anon_user_info_emoji_code: data.anon_user_info_emoji_code,
+                    anon_user_info_emoji_name: data.anon_user_info_emoji_name,
+                    member: member,
+                    message: message,
+                }, {
+                    onSuccess: (data) => {
+                        if (data) {
+                            setIsLoading(false);
+                            router.push('/message-sent');
+                        }
+                        localStorage.removeItem(MessageEnum.tempMessage);
+                        localStorage.removeItem(MessageEnum.targetUser);
+                    },
+                    onError: (err) => {
+                        setIsLoading(false);
+                        console.error(err)
+                        toast('We failed to send your message', {
+                            autoClose: 3000,
+                            type: 'error',
+                        })
+                    }
+                })
+            },
+            onError: (err) => {
+                setIsLoading(false);
+                console.error(err)
+                toast('We failed to send your message', {
+                    autoClose: 3000,
+                    type: 'error',
+                })
+            }
+        })
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `sendAnonymousMessage` function seems to be doing a lot of work. It's handling the generation of anonymous user info, initializing chat, and managing error states. This could potentially violate the Single Responsibility Principle (SRP) which suggests that a function should do one thing only.

Consider breaking down this function into smaller ones, each with its own responsibility. For example, you could have separate functions for generating anonymous user info, initializing chat, and handling errors. This would make your code more modular, easier to read, and maintain.

```diff
-    const sendAnonymousMessage = (anonymousToken: string) => {
-        const member = localStorage.getItem(MessageEnum.targetUser);
-        const message = localStorage.getItem(MessageEnum.tempMessage);
-        apiAnonymous.defaults.headers.common['Authorization'] = `Bearer ${anonymousToken}`;
-        generateAnonUser.mutate({ userId: member }, {
-            onSuccess: (data) => {
-                initChatAnon.mutate({
-                    anon_user_info_color_code: data.anon_user_info_color_code,
-                    anon_user_info_color_name: data.anon_user_info_color_name,
-                    anon_user_info_emoji_code: data.anon_user_info_emoji_code,
-                    anon_user_info_emoji_name: data.anon_user_info_emoji_name,
-                    member: member,
-                    message: message,
-                }, {
-                    onSuccess: (data) => {
-                        if (data) {
-                            setIsLoading(false);
-                            router.push('/message-sent');
-                        }
-                        localStorage.removeItem(MessageEnum.tempMessage);
-                        localStorage.removeItem(MessageEnum.targetUser);
-                    },
-                    onError: (err) => {
-                        setIsLoading(false);
-                        console.error(err)
-                        toast('We failed to send your message', {
-                            autoClose: 3000,
-                            type: 'error',
-                        })
-                    }
-                })
-            },
-            onError: (err) => {
-                setIsLoading(false);
-                console.error(err)
-                toast('We failed to send your message', {
-                    autoClose: 3000,
-                    type: 'error',
-                })
-            }
-        })
-    }
+    const generateAnonUserInfo = (anonymousToken: string, member: string) => {
+        apiAnonymous.defaults.headers.common['Authorization'] = `Bearer ${anonymousToken}`;
+        return generateAnonUser.mutate({ userId: member });
+    }
+
+    const initChatWithAnonUser = (data: any, member: string, message: string) => {
+        return initChatAnon.mutate({
+            anon_user_info_color_code: data.anon_user_info_color_code,
+            anon_user_info_color_name: data.anon_user_info_color_name,
+            anon_user_info_emoji_code: data.anon_user_info_emoji_code,
+            anon_user_info_emoji_name: data.anon_user_info_emoji_name,
+            member: member,
+            message: message,
+        });
+    }
+
+    const handleError = (err: any) => {
+        setIsLoading(false);
+        console.error(err)
+        toast('We failed to send your message', {
+            autoClose: 3000,
+            type: 'error',
+        })
+    }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->